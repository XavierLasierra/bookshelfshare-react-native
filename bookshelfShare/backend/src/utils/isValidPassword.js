const bcrypt = require("bcrypt");

function isValidPassword(password) {
  return bcrypt.compareSync(password, this.password);
}

module.exports = { isValidPassword };
