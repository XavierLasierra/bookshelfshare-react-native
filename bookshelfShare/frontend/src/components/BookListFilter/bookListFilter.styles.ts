import { StyleSheet } from 'react-native';
import stylesConstants from '../../styles/styles.constants';

export default StyleSheet.create({
  listTitle: {
    fontSize: stylesConstants.fontSize.big,
    fontWeight: 'bold',
    marginBottom: 10,
    textAlign: 'center'
  },
  inputContainer: {
    width: 300,
    height: 50
  },
  bookListFilterContainer: {
    alignItems: 'center',
    paddingVertical: stylesConstants.padding.small

  },
  input: {
    height: '100%',
    paddingHorizontal: 20,
    fontSize: stylesConstants.fontSize.mid,
    backgroundColor: stylesConstants.colors.white,
    borderRadius: 10,
    borderWidth: 1,
    borderColor: stylesConstants.colors.secondary
  }
});
