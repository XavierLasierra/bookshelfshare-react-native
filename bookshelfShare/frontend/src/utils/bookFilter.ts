export function bookFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfListFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.bookData.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfLocationFilter(filterLocation: number[], books: any) {
  return books.filter(({ customInformation: { location } }: any) => JSON.stringify(location)
  === JSON.stringify(filterLocation));
}
