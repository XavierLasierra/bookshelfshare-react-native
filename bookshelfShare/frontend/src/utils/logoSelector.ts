import BookIcon from '../assets/bookIcon.svg';
import SearchIcon from '../assets/searchIcon.svg';
import ProfileIcon from '../assets/profileIcon.svg';

export default function logoSelector(logoName: string) {
  switch (logoName) {
    case 'SearchIcon':
      return SearchIcon;
    case 'BookIcon':
      return BookIcon;
    case 'ProfileIcon':
      return ProfileIcon;
    default:
      return BookIcon;
  }
}
