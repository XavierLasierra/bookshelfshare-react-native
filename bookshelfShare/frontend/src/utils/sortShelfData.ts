export default function sortShelfData(shelf: any) {
  return shelf?.books.reduce((acc: any, book: any) => {
    const foundShelfBox = acc.find(({ location }: any) => JSON.stringify(location)
         === JSON.stringify(book?.customInformation?.location));
    if (foundShelfBox) {
      foundShelfBox.books.push(book.bookData);
    } else {
      acc.push({
        location: book?.customInformation?.location,
        books: [book.bookData]
      });
    }
    return acc;
  }, []);
}
