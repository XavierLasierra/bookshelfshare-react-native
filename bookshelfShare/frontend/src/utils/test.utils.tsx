import React from 'react';
import { render as rtlRender } from '@testing-library/react-native';
import { Provider } from 'react-redux';

import configureStore from '../redux/store';

function render(
  component: any,
  initialState: any
) {
  const store = configureStore(initialState);

  function Wrapper({ children }: any) {
    return (
      <Provider store={store}>
        {children}
      </Provider>
    );
  }
  return rtlRender(component, { wrapper: Wrapper });
}

// re-export everything
export * from '@testing-library/react-native';
// override render method
export { render };
