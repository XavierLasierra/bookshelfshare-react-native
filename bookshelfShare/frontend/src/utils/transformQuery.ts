interface Query {
    isbn?: string,
    inauthor?: string,
    intitle?: string,
    inpublisher?: string
}

export default function transformQuery(query: Query):string {
  const queryEntries = Object.entries(query);
  const url = queryEntries
    .reduce((acc, queryElement, index) => `${acc}${queryElement[0]}=${queryElement[1]}${index < queryEntries.length - 1 ? '+' : ''}`, '');
  return `${url}&key=${process.env.GOOGLE_API_KEY}`;
}
