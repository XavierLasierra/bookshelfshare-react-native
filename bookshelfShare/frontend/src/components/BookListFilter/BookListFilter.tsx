import React, { useState } from 'react';
import { Text, TextInput, View } from 'react-native';
import { IBookListFilterProps } from '../../types/interfaces';

import { bookFilter } from '../../utils/bookFilter';

import styles from './bookListFilter.styles';

export default function BookListFilter({
  listName,
  books,
  setFilteredBooks
}: IBookListFilterProps) {
  const [filter, setFilter] = useState('');

  function handleFilter(text: string) {
    setFilteredBooks(bookFilter(text, books));
    setFilter(text);
  }

  return (
    <View style={styles.bookListFilterContainer}>
      <Text style={styles.listTitle}>{listName.toUpperCase()}</Text>
      <View style={styles.inputContainer}>
        <TextInput
          style={styles.input}
          placeholder="Filter by title/author/publisher/isbn"
          value={filter}
          onChangeText={handleFilter}
          maxLength={25}
          testID="filterInput"
        />
      </View>
    </View>
  );
}
