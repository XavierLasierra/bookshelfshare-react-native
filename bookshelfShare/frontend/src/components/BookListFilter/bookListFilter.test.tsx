import React from 'react';
import BookListFilter from './BookListFilter';
import { render, fireEvent } from '../../utils/test.utils';

describe('Given a BookListFilter component', () => {
  describe('When it is rendered', () => {
    let setFilteredBooks: any;
    let screen: any;
    beforeEach(() => {
      setFilteredBooks = jest.fn();
      screen = render(<BookListFilter listName="to read" books={[]} setFilteredBooks={setFilteredBooks} />);
    });

    test('Then should match the snapshot', () => {
      expect(screen).toMatchSnapshot();
    });

    describe('And filterInput is changed', () => {
      test('Then setFilteredBooks should have been called', () => {
        const filterInput = screen.getByTestId('filterInput');
        fireEvent.changeText(filterInput, 'a');

        expect(setFilteredBooks).toHaveBeenCalled();
      });
    });
  });
});
