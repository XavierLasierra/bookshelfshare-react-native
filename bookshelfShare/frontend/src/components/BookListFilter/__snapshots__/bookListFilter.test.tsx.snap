// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a BookListFilter component When it is rendered Then should match the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "paddingVertical": 10,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
        "marginBottom": 10,
        "textAlign": "center",
      }
    }
  >
    TO READ
  </Text>
  <View
    style={
      Object {
        "height": 50,
        "width": 300,
      }
    }
  >
    <TextInput
      maxLength={25}
      onChangeText={[Function]}
      placeholder="Filter by title/author/publisher/isbn"
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#ECECEC",
          "borderRadius": 10,
          "borderWidth": 1,
          "fontSize": 16,
          "height": "100%",
          "paddingHorizontal": 20,
        }
      }
      testID="filterInput"
      value=""
    />
  </View>
</View>
`;
